Microsoft.IoT.Lightning.Providers initial commit

MahmoudGSaleh opened the pull request.

MahmoudGSaleh committed:
Initial Commit for Microsoft.IoT.Lightning.Providers library and samples

MahmoudGSaleh committed:
Update README.md

msftclas commented:
Hi **@msalehmsft**, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

MahmoudGSaleh committed:
Update SDK version to 10.0.1556.0

bfjelds commented on Microsoft.IoT.Lightning.Providers/Blinky/Blinky.csproj:
```diff
@@ -0,0 +1,152 @@
+    <DefaultLanguage>en-US</DefaultLanguage>
+    <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
+    <TargetPlatformVersion>10.0.10556.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.10240.0</TargetPlatformMinVersion>
```
10240 is RTM, can the dmap providers be used with an RTM SDK???

bfjelds commented on Microsoft.IoT.Lightning.Providers/Blinky/Package.appxmanifest:
```diff
@@ -0,0 +1,54 @@
+
+  <Identity
+    Name="ce88ee3f-c2d3-4148-9fd4-73d6a075b6b5"
+    Publisher="CN=msaleh"
```
remove msaleh references

bfjelds commented:
remove all project.lock.json files

bfjelds commented:
be sure to include all sln files (i didn't see one for any blinky\* projects)

bfjelds commented:
be sure you have LICENSE.txt in all appropriate places

bfjelds commented on Microsoft.IoT.Lightning.Providers/Blinky/MainPage.xaml.cs:
```diff
@@ -0,0 +1,141 @@
+﻿using System;
```
add copyright comment to all source code

ooeygui replied:
New guidance from LCA is that top level license.txt is all that is required - each file need not have a copyright and licence header.

MahmoudGSaleh committed:
Addressed PR review feedback for initial commit

- Removed references to SDK 10240, and added 10556, min needed for
deployment.
- Replaced all user references with Microsoft
- project.lock.json files removed
- Removed top level .sln file
- Added new .sln files, one for the provider library and for each of the
samples
- LICENSE.txt added everywhere, top level and in each of the solutions

MahmoudGSaleh commented:
Thanks! The PR now has the following changes:
1. Removed references to SDK 10240, and added 10556, min needed for deployment.
2. Replaced all user references with Microsoft
3. project.lock.json files removed
4. Removed top level .sln file
5. Added new .sln files, one for the provider library and for each of the samples
6. LICENSE.txt added everywhere, top level and in each of the solutions

MahmoudGSaleh committed:
Unify the way Lightning Provider is set as the default in all samples

In addition change the provider name to Lightning*Provider instead of
Provider. The naming convention applies to all provider types exported.

MahmoudGSaleh committed:
Update comments to indicate the inbox provider will be the default if Lighting is not enabled

bfjelds commented on Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj:
```diff
@@ -0,0 +1,267 @@
+  <ImportGroup Label="Shared">
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
```
what are all of these imports?

bfjelds commented on Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj.filters:
```diff
@@ -0,0 +1,194 @@
+    <ClCompile Include="Provider.cpp" />
+    <ClCompile Include="SpiDeviceProvider.cpp" />
+    <ClCompile Include="Dummy.cpp" />
+    <ClCompile Include="@(ArduinoSDKSources)" />
```
why are there 6 clcompile items for arduinosdksrouces?

MahmoudGSaleh replied:
Looks like they get added by VC++ when referencing these sources from the targets of the nuget packages.

bfjelds commented:
remove all project.lock.json files

bfjelds commented:
check with jesse, but for the sample aggregate provider i'm working on, jesse suggested that the only public constructor be the aggregate constructor.

MahmoudGSaleh committed:
Delete and ignore project.lock.json

MahmoudGSaleh commented:
Still waiting on the final SDK version to include.
